Classes/Player/PlayerController.swift

```diff
@@ -71,9 +77,14 @@ class PlayerController: NSObject, Player {
     var shouldRefresh: Bool = false
     
     func prepare(_ config: MediaConfig) {
-        // configure media sources content request adapter
-        if let contentRequestAdapter = self.contentRequestAdapter {
-            config.mediaEntry.configureMediaSource(withContentRequestAdapter: contentRequestAdapter)
+        // configure media sources content request adapter if request adapter exists
+        if let _ = self.contentRequestAdapter {
```

Comment:
KalturaPlaybackRequestAdapter must be created only once, in PlayerLoader. Then, PlayerController only knows the protocol, not the implementation.
Add updateParams(player) in KalturaPlaybackRequestAdapter (and in PKRequestParamsAdapter) so that the sessionId can be updated.

Consider that an app may provide its own implementation of PKRequestParamsAdapter -- in Settings. You can't override it.